Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsProvider.cs

```diff
@@ -145,6 +145,17 @@ public override void OnGUI(string searchContext)
                 EditorGUILayout.Space();
                 EditorGUILayout.PropertyField(m_EditorInputBehaviorInPlayMode, m_EditorInputBehaviorInPlayModeContent);
 
+                EditorGUILayout.Space();
+                EditorGUILayout.LabelField("Shortcut support", EditorStyles.boldLabel);
+                EditorGUILayout.Space();
+                EditorGUILayout.PropertyField(m_ImprovedShortcutSupportEnabled, m_ImprovedShortcutSupportContent);
+
+                EditorGUILayout.HelpBox("Please note that enabling Improved Shortcut Support will cause actions to consume and block any other actions which are enabled and sharing the same controls. "
+                    + "E.g. when pressing the 'Shift+B' keys, the associated action would trigger but any action bound to just the 'B' key would be prevented from triggering at the same time, which would be the expected result in this example. "
+                    + "However, in other cases this might not give the desired result. E.g. if the 'W', 'A', 'S' and 'D' keys are bound to different actions which are all enabled at the same time, then only one of those actions will trigger. "
+                    + "Therefore if you enable this feature, be sure to remove any of these duplicate control bindings so that controls are only ever bound to a single action. Alternatively, you can disable actions or action maps at runtime to remove these conflicts."
+                    , MessageType.None);
```

Comment:
Some suggestions:

- "Please note that enabling Improved Shortcut Support will cause actions _with composite bindings_ to consume _input_ and block..."
- Could point out that blocking occurs even across action maps. I don't think users are really often aware of where the actions live and just enable entire assets.
- I feel we can leave out the repeated instruction about Shift-B and B since that is more about the features intended purpose and is already in the tooltip.
- Would this benefit from a more detailed worked example describing two action maps, one for UI and one for Gameplay, where both use the up, down, left, and right keys (pretty common one on the forums) and maybe some other action that will be lower priority and only uses left and right keys for example.
- It would help users to know how priority is decided between actions. Should we include some instructions on that, i.e. number of composite parts determines priority but actions with the same number of parts are non-deterministic?
- Is the advice really to remove duplicate controls? I like the second bit about disabling actions but maybe we could expand it to something like "Since event consumption only occurs for enabled actions, you can resolve most unexpected issues by ensuring that only those action maps that are relevant to your game's current context are enabled. Enable or disable actions as your game or application moves between different contexts."

Reply:
Tried to reword this incorporating your suggestions